scio-core/src/main/scala/com/spotify/scio/coders/instances/ScalaCoders.scala

```diff
@@ -335,6 +336,75 @@ private[coders] class MapCoder[K, V](val kc: BCoder[K], val vc: BCoder[V])
   override def getCoderArguments: JList[_ <: BCoder[_]] = List(kc, vc).asJava
 }
 
+private[coders] class SortedMapCoder[K: Ordering, V](
```

Comment:
Now we have several `MapCoder`s, let's factorize some code in a `MapLikeCoder` abstract class